Fabric.Identity.API/scripts/CatalystDosIdentity.psm1

```diff
@@ -127,12 +127,20 @@ function Get-ClientRegistration {
 function New-ClientRegistration {
     param(
         [Parameter(Mandatory = $True)] [Uri] $identityUrl,
-        [Parameter(Mandatory = $True)] [string] $body,
+        [Parameter(Mandatory = $True)] $body,
         [Parameter(Mandatory = $True)] [string] $accessToken
     )
 
     [Uri] $url = "$($identityUrl.OriginalString.TrimEnd("/"))/api/client"
 
+    if (!($body -is [string])) {
+        $clientObject = $body
+        $body = $body | ConvertTo-Json
```

Comment:
Is it possible that ConvertTo-Json can throw an error. This is what I have seen can happen in forums. ConvertTo-Json : The converted JSON string is in bad format. I guess if that error happens it is pretty clear, it might not need to go threw a try catch.

Reply:
I would view that error as a parameter error, so I think it's ok to bypass the try/catch